### PR TITLE
docker-compose update - Closes #786

### LINKED ISSUE #786: Update docker configs
### Description

Adjust docker-compose files and configs to ensure Lisk Service is working flawlessly in Docker.

Update Redis configs so it guarantees stability on the deployed server.

Environment files need to be adjusted and reduced to one file (placed in `docker/sample.env`).

Other setups can be applied as sample files.

### Acceptance Criteria

- Dockerfiles are updated
- Redis configs are updated
- Environment files are updated
- Readme is updated

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?

This PR resolves #786

### How was it solved?

- [x] Rewritten docker-compose file
- [x] Updated Redis configurations
- [x] Updated Makefile
- [x] Updated Readme

### How was it tested?

Jenkins + Linux + Local

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
-up-core3: up-custom-core3-default
-
-up-custom-%:
-	cd ./docker && $(compose) --env-file ./network/$*.env up --detach
+up:
+	docker-compose up -d
 
 down:
-	cd ./docker && $(compose) down --volumes --remove-orphans
+	$(compose) down --volumes --remove-orphans
```
Comment (sameersubudhi) [inline Makefile:10]: I would also recommend including a `restart` option:

```
restart:
	$(compose) restart
```

File: Makefile [hunk 2]
```diff
@@ -1,30 +1,16 @@
...
-up-mainnet: up-custom-core2-mainnet
-
-up-testnet: up-custom-core2-testnet
-
-up-core3: up-custom-core3-default
-
-up-custom-%:
-	cd ./docker && $(compose) --env-file ./network/$*.env up --detach
+up:
+	docker-compose up -d
```
Comment (sameersubudhi) [inline Makefile:7]: For consistency purposes:
```suggestion
	$(compose) up -d
```

File: README.md [hunk 3]
```diff
@@ -82,27 +82,27 @@ make build
...
+
+```bash
+nano .env # or vim
+```
 
-# Enable transaction statistics
-export ENABLE_TRANSACTION_STATS="true"
-export TRANSACTION_STATS_HISTORY_LENGTH_DAYS="366"
-export TRANSACTION_STATS_UPDATE_INTERVAL="3600"
+The example below is valid with the assumption the Lisk Core node is running on host machine, not in a Docker container.
```
Comment (sameersubudhi) [inline README.md:97]:
```suggestion
The example below is valid with the assumption the Lisk Core node is running on the host machine, not inside of a Docker container.
```

File: README.md [hunk 4]
```diff
@@ -82,27 +82,27 @@ make build
...
-# Enable fee estimator
-export ENABLE_FEE_ESTIMATOR_QUICK="true"
+```
+## Required
+# The local Lisk Core node WebSocket API port
+export LISK_CORE_WS="ws://host.docker.internal:8080"
 ```
 
-When running a local instance of Lisk Core and Lisk Service that is run by Docker, then the following variable needs to be set: `LISK_CORE_WS="ws://host.docker.internal:8080"`.
+When running a Docker instance of Lisk Core that is run by Docker, the variable needs to refer to the container: `LISK_CORE_WS="ws://<your_docker_container>:8080"`.
```
Comment (sameersubudhi) [inline README.md:105]:
```suggestion
When running Lisk Core inside of a Docker container, the variable needs to refer to the container: `LISK_CORE_WS="ws://<your_docker_container>:8080"`.
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find a few recommendations

File: docker/example.env [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+# Missing blocks job
+ENABLE_JOB_MISSING_BLOCKS=true
+INDEX_MISSING_BLOCKS_RANGE=0
+
+# Log debug messages
+#DEBUG=*dynamicFees*
+
+# # Lisk ecosystem configuration
+LISK_CORE_WS=ws://host.docker.internal:4004
```
Comment (sameersubudhi) [inline docker/example.env:46]: Please consider setting it to the default Lisk Core WS port
```suggestion
LISK_CORE_WS=ws://host.docker.internal:8080
```

File: README.md [hunk 6]
```diff
@@ -82,27 +82,27 @@ make build
...
-# The local Lisk Core node WebSocket API port
-export LISK_CORE_WS="ws://localhost:8080"
 
-## Optional
-# To index all blocks in the blockchain (might take a while)
-export INDEX_N_BLOCKS="0"
+In the next step, set the required environment variables.
+
+```bash
+nano .env # or vim
```
Comment (fchavant) [inline README.md:94]:
```suggestion
$EDITOR .env
```

File: README.md (hunk 3 above)
Comment (fchavant) [reply README.md:97]: "The example assumes that [...]"?

File: README.md (hunk 4 above)
Comment (fchavant) [reply README.md:105]: And both containers need to have a network in common.

File: docker/example.env [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+# Lisk Service Core
+# SERVICE_CORE_REDIS=redis://redis_core_persistent:6379/0
+# SERVICE_CORE_REDIS_VOLATILE=redis://redis_core_volatile:6379/0
+# SERVICE_CORE_MYSQL=mysql://lisk:password@mysql_core:3306/lisk
+
+# Lisk static assets, ie. known account lists
+LISK_STATIC=https://static-data.lisk.com
+
+# Lisk Service geolocation backend
```
Comment (fchavant) [inline docker/example.env:21]: Does empty mean disabled?

File: docker/redis.persistent.conf [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+tcp-keepalive 300
+
+# Snapshotting
+appendonly yes
+aof-use-rdb-preamble yes
+save ""
+
+# Memory management
+maxmemory 512mb
+maxmemory-policy volatile-lru
```
Comment (sameersubudhi) [inline docker/redis.persistent.conf:17]: For this policy to be effective, won't we also need to ensure that all the DB entries have the expiry (TTL) set?

Ref: https://redis.io/topics/lru-cache
```
volatile-lru: evict keys by trying to remove the less recently used (LRU) keys first, but only among keys that have an expire set, in order to make space for the new data added.
```

File: docker/example.env (hunk 7 above)
Comment (MichalTuleja) [reply docker/example.env:21]: Yes, it is equivalent to that.

Comment (fchavant) [reply docker/example.env:21]: Maybe the comment should mention that then?

File: docker/redis.persistent.conf (hunk 8 above)
Comment (MichalTuleja) [reply docker/redis.persistent.conf:17]: Yes, only keys with expiry time will be removed, other keys will be persistent.

File: README.md [hunk 9]
```diff
@@ -124,6 +124,8 @@ To stop the application execute the following command:
 make down
 ```
 
+> Optional: It is possible to use regular docker-compose commands such as `docker-compose up -d`. Please check the `Makefile` for more examples.
+
 ## Benchmark
 
 Assuming lisk-service is running on localhost:9901, and you are in the root of this repo, you can run:
```
Comment (chris529) [inline README.md:131]:
```suggestion
Assuming lisk-service is running on the localhost:9901, and you are in the root of this repo, you can run the following:
```

File: docker/redis.persistent.conf [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+#
+# This config enables persistence
+bind 0.0.0.0
+port 6379
+tcp-backlog 127
+timeout 0
+tcp-keepalive 300
+
+# Snapshotting
+appendonly yes
```
Comment (chris529) [inline docker/redis.persistent.conf:11]:
```suggestion
append-only yes
```

**Review by chris529 — COMMENTED** [review]
Looks good, just a few minor points 👍

File: docker/redis.persistent.conf (hunk 10 above)
Comment (MichalTuleja) [reply docker/redis.persistent.conf:11]: This setting comes from a 3rd party software, it is not possible to change it.

File: Makefile (hunk 1 above)
Comment (MichalTuleja) [reply Makefile:10]: Added `restart`

File: docker/example.env (hunk 7 above)
Comment (MichalTuleja) [reply docker/example.env]: Added explanation what empty means.

**Review by fchavant — APPROVED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Pull request merged**